### PR TITLE
Do not overwrite an existing `error.cause` value.

### LINKED ISSUE #48: Originating `error.cause` is being ignored
Nodejs `undici` can throw an error with a `cause` property:

https://github.com/nodejs/node/blob/668437ccada33c03bb454a19c61b030c028076b0/deps/undici/src/lib/fetch/index.js#L225-L230

In the `_handleClientError` helper, `error.cause` is being overwritten without first checking to see if it already exists:
https://github.com/digitalbazaar/webkms-client/blob/ab87f0b708f4f18bfa63e0afe3c599a5eebfbd27/lib/KmsClient.js#L782-L802

In addition to the original `cause` being lost, `cause` is logged as `$ref: $`
```
error: {
cause: {
$ref: "$"
}
message: "WebKMS client error: fetch failed."
requestUrl: "https://kms.vereskms.com/kms/keystores/z1AG7ybSKVmAWCbYwR34f8H2J/keys/z1A2Cnt8u9ciWsmt2RNkW15ot"
stack: "TypeError: WebKMS client error: fetch failed.
    at Object.fetch (node:internal/deps/undici/undici:11576:11)
    at async fn (file:///home/node/app/node_modules/ky/distribution/core/Ky.js:18:28)
    at async _handleResponse (file:///home/node/app/node_modules/@digitalbazaar/http-client/lib/httpClient.js:101:16)
    at async KmsClient.sign (file:///home/node/app/node_modules/@digitalbazaar/webkms-client/lib/KmsClient.js:436:22)
    at async AsymmetricKey.sign (file:///home/node/app/node_modules/@digitalbazaar/webkms-client/lib/AsymmetricKey.js:84:28)
    at async Ed25519Signature2020.sign (file:///home/node/app/node_modules/@bedrock/vc-issuer/node_modules/@digitalbazaar/ed25519-signature-2020/lib/Ed25519Signature2020.js:82:28)
    at async Ed25519Signature2020.createProof (/home/node/app/node_modules/jsonld-signatures/lib/suites/LinkedDataSignature.js:151:13)
    at async ProofSet.add (/home/node/app/node_modules/jsonld-signatures/lib/ProofSet.js:63:19)
    at async Object.sign (/home/node/app/node_modules/jsonld-signatures/lib/jsonld-signatures.js:118:12)
    at async issue (file:///home/node/app/node_modules/@bedrock/vc-issuer/lib/issuer.js:92:28)"
}
level: "error"
module: "bedrock-vc-issuer"
workerId: "845c224aabff05a9"
workerPid: 39
}
@message: "WebKMS client error: fetch failed."
@timestamp: "2023-09-12T08:23:19.305Z"
}
```

### PR DESCRIPTION
**Pull request by mattcollier**
Fixes #48

### CODE REVIEW

**Review by dlongley — CHANGES_REQUESTED** [review]
The `_handleClientError` function seems a little off ... and I'm worried we might be trading one bug for another / leaving other bugs here. See suggestion.

File: lib/KmsClient.js [hunk 1]
```diff
@@ -789,15 +789,15 @@ function _handleClientError({
     error = new Error(`${errorMessage}: ${notFoundMessage}`);
     error.status = 404;
   } else {
-    error = cause;
-    error.message = `WebKMS client error: ${cause.message}`;
+    error = new Error(`WebKMS client error: ${errorMessage}`);
   }
```
Comment (dlongley) [inline lib/KmsClient.js:793]: I recommend we do this for some backwards compatibility as well:

```suggestion
    error = new Error(`WebKMS client error: ${errorMessage}`);
    error.status = cause.status;
  }
```

**Review by dlongley — APPROVED** [review]
Thanks! Approving w/given suggestion.

File: lib/KmsClient.js (hunk 1 above)
Comment (mattcollier) [reply lib/KmsClient.js:793]: @dlongley do you think we should test that `cause.status` exists? to avoid `error.status = undefined`.

Comment (dlongley) [reply lib/KmsClient.js:793]: Sure, let's add that too.

**Pull request merged**